contracts/pair_stable/src/contract.rs

```diff
@@ -454,8 +454,40 @@ pub fn provide_liquidity(
             return Err(ContractError::LiquidityAmountTooSmall {});
         }
 
+        // Get fee info from the factory
+        let fee_info = query_fee_info(
+            &deps.querier,
+            &config.factory_addr,
+            config.pair_info.pair_type.clone(),
+        )
+        .unwrap_or_default(); // There may no fee exist thus 0 is a default fee.
+
+        let fee = fee_info.total_fee_rate / Uint128::from(2u8);
```

Comment:
Let's try do not use magic numbers, instead use constant if it possible. Or maybe add some comment above what they are means, math formula or additional description why we are use '2'.